support/BRSyncMode.c

```diff
@@ -87,11 +92,76 @@ BRDisconnectReasonPosix(int errnum) {
 }
 
 extern const char *
-BRDisconnectReasonPosixGetMessage(BRDisconnectReason *reason) {
-    assert (reason->type == DISCONNECT_REASON_POSIX);
-    return strerror (reason->u.posix.errnum);
+BRDisconnectReasonGetMessage(BRDisconnectReason *reason) {
+    char *message = NULL;
+
+    switch (reason->type) {
+        case DISCONNECT_REASON_POSIX: {
+            if (NULL != (message = strerror (reason->u.posix.errnum))) {
+                message = strdup (message);
+            }
+            break;
+        }
+        default: {
+            break;
+        }
+    }
+
+    return message;
 }
 
+/// MARK: Sync Stopped Reason
+
+extern BRSyncStoppedReason
+BRSyncStoppedReasonComplete(void) {
+    return (BRSyncStoppedReason) {
+        SYNC_STOPPED_REASON_COMPLETE
+    };
+}
+
+extern BRSyncStoppedReason
+BRSyncStoppedReasonRequested(void) {
+    return (BRSyncStoppedReason) {
+        SYNC_STOPPED_REASON_REQUESTED
+    };
+}
+
+extern BRSyncStoppedReason
+BRSyncStoppedReasonUnknown(void) {
+    return (BRSyncStoppedReason) {
+        SYNC_STOPPED_REASON_UNKNOWN
+    };
+}
+
+extern BRSyncStoppedReason
+BRSyncStoppedReasonPosix(int errnum) {
+    return (BRSyncStoppedReason) {
+        SYNC_STOPPED_REASON_POSIX,
+        { .posix = { errnum } }
+    };
+}
+
+extern const char *
+BRSyncStoppedReasonGetMessage(BRSyncStoppedReason *reason) {
```

Comment:
Since this has a `strdup()` I think the return can't be `const`.  Also is worth noting that the return can be `NULL` - like in a documentation comment.  [Perhaps it is time for `\#define MIGHT_BE_NULLABLE` (name, obviously, TBD).]

Or don't `strdup()`.

Reply:
Same with `BRDisconnectReasonGetMessage()`, `BRTransferSubmitErrorGetMessage()`, ...